hebsafeharbor/identifier/signals/hebrew_city_recognizer.py

```diff
@@ -57,27 +59,138 @@ def analyze(
-            adjusted_score = self.DEFAULT_CONFIDENCE_LEVEL + (overlap_ratio * self.LOCATION_OVERLAP_FACTOR)
             result = RecognizerResult(
                 entity_type="CITY",
                 start=term_start,
                 end=term_start + term_len,
-                score=adjusted_score,
-                analysis_explanation=AnalysisExplanation(self.name, adjusted_score),
+                score=self.DEFAULT_CONFIDENCE_LEVEL,
+                analysis_explanation=AnalysisExplanation(self.name, self.DEFAULT_CONFIDENCE_LEVEL),
                 recognition_metadata={RecognizerResult.RECOGNIZER_NAME_KEY: self.name},
             )
 
             results.append(result)
 
+        results = self.enhance_using_other_entities(nlp_artifacts, results)
+
+        return results
+
+    def enhance_using_context(
+        self,
+        text: str,
+        raw_recognizer_results: List[RecognizerResult],
+        other_raw_recognizer_results: List[RecognizerResult],
+        nlp_artifacts: NlpArtifacts,
+        context: Optional[List[str]] = None,
+    ) -> List[RecognizerResult]:
+        """Overrides generic context enhancer with recognizer-specific.
+        Enhance confidence score using context of the entity.
+
+        in case a result score is boosted, derived class need to update
+        result.recognition_metadata[RecognizerResult.IS_SCORE_ENHANCED_BY_CONTEXT_KEY]
+
+        :param text: The actual text that was analyzed
+        :param raw_recognizer_results: This recognizer's results, to be updated
+        based on recognizer specific context.
+        :param other_raw_recognizer_results: Other recognizer results matched in
+        the given text to allow related entity context enhancement
+        :param nlp_artifacts: The nlp artifacts contains elements
+                              such as lemmatized tokens for better
+                              accuracy of the context enhancement process
+        :param context: list of context words
+        """
+
+        for result in raw_recognizer_results:
+            result.recognition_metadata[RecognizerResult.IS_SCORE_ENHANCED_BY_CONTEXT_KEY] = True
+            word = text[result.start:result.end]
+
+            sentence = self.__extract_sentence_containing_position(
+                nlp_artifacts=nlp_artifacts, start=result.start
+            )
+
+            if sentence != "":
+                supportive_context = self.__find_supportive_context_in_sentence(
+                    sentence, word
+                )
+                if supportive_context:
+                    result.score += self.CONTEXT_ENHANCEMENT_FACTOR
+
+                    # Update the explainability object with context information
+                    # helped improving the score
+                    result.analysis_explanation.set_supportive_context_word(
+                        supportive_context
```

Comment:
`supportive_context` is a list of strings, but the `set_supportive_context_word` expects one string.

Reply:
resolved